packages/wonder-blocks-modal/src/components/flexible-dialog.tsx

```diff
@@ -0,0 +1,225 @@
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+import type {PropsFor, StyleType} from "@khanacademy/wonder-blocks-core";
+
+import {Id, View} from "@khanacademy/wonder-blocks-core";
+import {breakpoint} from "@khanacademy/wonder-blocks-tokens";
+import {Heading} from "@khanacademy/wonder-blocks-typography";
+import ModalDialog from "./modal-dialog";
+import FlexiblePanel, {BackgroundStyles} from "./flexible-panel";
+
+/**
+ * The visible title content for the dialog. An ID will be automatically applied
+ * for aria-labelledby on the dialog to the heading content or a div wrapper.
+ */
+type TitleAndAriaLabelledBy = {
+    title: React.ReactNode;
+    "aria-label"?: never;
+    "aria-labelledby"?: string;
+};
+/**
+ * The optional accessible name of the modal.
+ * This is useful when there is no main heading in the dialog. A visible
+ * heading with an id is preferred for automatic matching with aria-labelledby.
+ */
+type AriaLabelOnly = {
+    title?: never;
+    "aria-label": string;
+    "aria-labelledby"?: never;
+};
+/**
+ * The optional ID of the content describing this dialog, if applicable.
+ */
+type AriaLabelledByOnly = {
+    title?: never;
+    "aria-label"?: never;
+    "aria-labelledby": string;
+};
+
+/**
+ * One of the labeling methods is required:
+ */
+type AccessibleLabelProps =
```

Comment:
I recently learned we have `*.typestest.tsx` tests in WB for TypeScript types! It's helpful for asserting when there should/shouldn't be a TS errors based on prop combinations. I think it'd be helpful for this component too! I did something similar for [badges](https://github.com/Khan/wonder-blocks/blob/main/packages/wonder-blocks-badge/src/components/__tests__/badge.typestest.tsx)!